pkg/cluster/arooperator_test.go

```diff
@@ -0,0 +1,297 @@
+			},
+			mocks: func(dep *mock_deploy.MockOperator) {
+				dep.EXPECT().
+					CreateOrUpdate(gomock.Any()).
+					Return(nil)
+			},
+			kubernetescli: fake.NewSimpleClientset(),
+		},
+		{
+			name: "create/update failure",
+
+			fixtureChecker: func(fixture *testdatabase.Fixture, checker *testdatabase.Checker, dbClient *cosmosdb.FakeOpenShiftClusterDocumentClient) {
+				doc := &api.OpenShiftClusterDocument{
+					Key: strings.ToLower(key),
+					OpenShiftCluster: &api.OpenShiftCluster{
+						ID: key,
+						Properties: api.OpenShiftClusterProperties{
+							IngressProfiles: []api.IngressProfile{
+								{
+									Visibility: api.VisibilityPublic,
+									Name:       "default",
+								},
+							},
+							ProvisioningState: api.ProvisioningStateAdminUpdating,
+							ClusterProfile: api.ClusterProfile{
+								Version: "4.8.18",
+							},
+							NetworkProfile: api.NetworkProfile{
+								APIServerPrivateEndpointIP: "1.2.3.4",
+							},
+						},
+					},
+				}
+				fixture.AddOpenShiftClusterDocuments(doc)
+				doc.Dequeues = 1
+				checker.AddOpenShiftClusterDocuments(doc)
+			},
+			mocks: func(dep *mock_deploy.MockOperator) {
+				dep.EXPECT().
+					CreateOrUpdate(gomock.Any()).
+					Return(errors.New("Mock return: CreateFailed"))
+			},
+
+			kubernetescli: fake.NewSimpleClientset(),
+			wantErr:       "Mock return: CreateFailed",
+		},
+		{
+			name: "enriched data not available - skip",
+			fixtureChecker: func(fixture *testdatabase.Fixture, checker *testdatabase.Checker, dbClient *cosmosdb.FakeOpenShiftClusterDocumentClient) {
+				doc := &api.OpenShiftClusterDocument{
+					Key: strings.ToLower(key),
+					OpenShiftCluster: &api.OpenShiftCluster{
+						ID: key,
+						Properties: api.OpenShiftClusterProperties{
+							IngressProfiles:   nil,
+							ProvisioningState: api.ProvisioningStateAdminUpdating,
+						},
+					},
+				}
+				fixture.AddOpenShiftClusterDocuments(doc)
+
+				doc.Dequeues = 1
+				checker.AddOpenShiftClusterDocuments(doc)
+			},
+			kubernetescli: fake.NewSimpleClientset(),
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			controller := gomock.NewController(t)
+			defer controller.Finish()
+
+			dep := mock_deploy.NewMockOperator(controller)
+			if tt.mocks != nil {
+				tt.mocks(dep)
+			}
+
+			dbOpenShiftClusters, dbClient := testdatabase.NewFakeOpenShiftClusters()
+			fixture := testdatabase.NewFixture().WithOpenShiftClusters(dbOpenShiftClusters)
+			checker := testdatabase.NewChecker()
+
+			if tt.fixtureChecker != nil {
+				tt.fixtureChecker(fixture, checker, dbClient)
+			}
+
+			err := fixture.Create()
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			doc, err := dbOpenShiftClusters.Dequeue(ctx)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			m := &manager{
+				log:                 logrus.NewEntry(logrus.StandardLogger()),
+				doc:                 doc,
+				db:                  dbOpenShiftClusters,
+				kubernetescli:       tt.kubernetescli,
```

Comment:
I think we can drop this line as we are not using kube client directly and mock the operator deployer.

Same for the test below.

```suggestion
```

Reply:
Fixed